src/libsystemd/sd-device/device-enumerator.c

```diff
@@ -223,49 +223,64 @@ int device_enumerator_add_match_is_initialized(sd_device_enumerator *enumerator)
         return 1;
 }
 
-static int device_compare(sd_device * const *_a, sd_device * const *_b) {
-        sd_device *a = *(sd_device **)_a, *b = *(sd_device **)_b;
-        const char *devpath_a, *devpath_b, *sound_a;
-        bool delay_a, delay_b;
-        int r;
+static int sound_device_compare(const char *a, const char *b) {
```

Comment:
it's not clear these params are the devpath of these devices. maybe add comment (or rename params?)